### PR TITLE
feat: update gatsby config for chinese

### LINKED ISSUE #46202: Update learn client app logic
Details: 

From https://github.com/freeCodeCamp/freeCodeCamp/issues/45623#issuecomment-1096798841

Within the client:

- https://github.com/freeCodeCamp/freeCodeCamp/blob/main/client/src/components/create-language-redirect.ts (handles the logic for `chinese.freecodecamp.org` instead of `freecodecamp.org/chinese`. #46220
- https://github.com/freeCodeCamp/freeCodeCamp/blob/main/client/src/utils/algolia-locale-setup.ts (points to chinese news) #46222
- https://github.com/freeCodeCamp/freeCodeCamp/blob/main/client/utils/tags.tsx (injects specific tags for chinese) #46222
- https://github.com/freeCodeCamp/freeCodeCamp/blob/main/client/gatsby-config.js (gatsby config has logic for english and chinese to not use language subpaths) #46424

Some of the `i18n` files in the client have links which point to `chinese.freecodecamp.org`.

--

From https://github.com/freeCodeCamp/freeCodeCamp/issues/45623#issuecomment-1097726605

Tests and mocks:
* https://github.com/freeCodeCamp/freeCodeCamp/blob/main/client/src/__mocks__/gatsby.js #46424
* https://github.com/freeCodeCamp/freeCodeCamp/blob/main/client/src/components/create-language-redirect.test.ts #46220 
* https://github.com/freeCodeCamp/freeCodeCamp/blob/main/client/utils/tags.test.tsx #46222

One challenge file:
* https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/chinese/02-javascript-algorithms-and-data-structures/basic-javascript/return-early-pattern-for-functions.md

Cors config:
* https://github.com/freeCodeCamp/freeCodeCamp/blob/main/config/cors-settings.js

Docs:

* https://github.com/freeCodeCamp/freeCodeCamp/blob/main/docs/devops.md #46207
* https://github.com/freeCodeCamp/freeCodeCamp/blob/main/docs/how-to-test-translations-locally.md #46207
* https://github.com/freeCodeCamp/freeCodeCamp/blob/main/docs/index.md #46207

### PR DESCRIPTION
**Pull request by naomi-lgbt**
Checklist:

- [X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #46202 https://github.com/freeCodeCamp/freeCodeCamp/issues/40867

Didn't see a PR for these ones.

### CODE REVIEW

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Review by ShaunSHamilton — APPROVED** [review]

**Review by Sboonny — APPROVED** [review]

File: client/gatsby-config.js [hunk 1]
```diff
@@ -10,10 +10,7 @@ const { clientLocale, curriculumLocale, homeLocation, sentryClientDSN } =
   envData;
 
 const curriculumIntroRoot = path.resolve(__dirname, './src/pages');
-const pathPrefix =
-  clientLocale === 'english' || clientLocale === 'chinese'
-    ? ''
-    : '/' + clientLocale;
+const pathPrefix = clientLocale === 'english' ? '' : '/' + clientLocale;
```
Comment (Sboonny) [inline client/gatsby-config.js:13]:
```suggestion
let pathPrefix = ''
if (clientLocale !== 'english') pathPrefix = `/${clientLocale}`;
```

File: client/src/__mocks__/gatsby.ts [hunk 2]
```diff
@@ -19,10 +19,7 @@ module.exports = {
       })
   ),
   withPrefix: jest.fn().mockImplementation((path: string) => {
-    const pathPrefix =
-      clientLocale === 'english' || clientLocale === 'chinese'
-        ? ''
-        : '/' + clientLocale;
+    const pathPrefix = clientLocale === 'english' ? '' : '/' + clientLocale;
```
Comment (Sboonny) [inline client/src/__mocks__/gatsby.ts:22]:
```suggestion
let pathPrefix = ''
if (clientLocale !== 'english') pathPrefix = `/${clientLocale}`;
```

**Review by Sboonny — COMMENTED** [review]
Because this is a simple logic, is it okay to use `if` statement for readability

**Comment by raisedadead** [comment]
> Because this is a simple logic, is it okay to use `if` statement for readability

Well, IDK. 

I find this is cosmetic; in fact, I think the ternary checks are more consistent all over the codebase, making it more readable?

**Comment by Sboonny** [comment]
True, but when I approved it first it took me a minute to get it. After I have received 2nd notification about it, it took me a minute again to understand it.

I think the issue: variable is assigned to the ternary operation, so it takes me a minute to correctly read it.

**Comment by Sboonny** [comment]
Let's merge them first, then we can discuss refactoring after

**Review by Sboonny — APPROVED** [review]
LGTM 🎉, thank you for your contribution to the page! 👍.

**Pull request merged**